cli/bpmetadata/parser/state_parser_test.go

```diff
@@ -0,0 +1,180 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func TestParseOutputTypesFromState_WithSimpleTypes(t *testing.T) {
+	t.Parallel()
+	stateData := []byte(`
+{
+  "version": 4,
+  "terraform_version": "0.12.0",
+  "serial": 8,
+  "lineage": "491fd8f4-81b5-9890-520c-8a173c36e483",
+  "outputs": {
+    "boolean_output": {
+      "value": true,
+      "type": "bool"
+    },
+    "number_output": {
+      "value": 42,
+      "type": "number"
+    },
+    "string_output": {
+      "value": "foo",
+      "type": "string"
+    }
+  },
+  "resources": []
+}
+`)
+	want := map[string]*structpb.Value{
+		"boolean_output": structpb.NewStringValue("bool"),
+		"number_output":  structpb.NewStringValue("number"),
+		"string_output":  structpb.NewStringValue("string"),
+	}
+	got, err := ParseOutputTypesFromState(stateData)
+	if err != nil {
+		t.Errorf("ParseOutputTypesFromState() error = %v", err)
+		return
+	}
+	if diff := cmp.Diff(got, want, cmp.Comparer(compareStructpbValues)); diff != "" {
+		t.Errorf("ParseOutputTypesFromState() mismatch (-got +want):\n%s", diff)
+	}
+}
+
+func TestParseOutputTypesFromState_WithComplexTypes(t *testing.T) {
+	t.Parallel()
+	stateData := []byte(`
+{
+  "version": 4,
+  "terraform_version": "0.12.0",
+  "serial": 8,
+  "lineage": "491fd8f4-81b5-9890-520c-8a173c36e483",
+  "outputs": {
+    "interpolated_deep": {
+      "value": {
+        "foo": "bar",
+        "map": {
+          "bar": "baz",
+          "id": "424881806176056736"
+        },
+        "number": 42
+      },
+      "type": [
+        "object",
+        {
+          "foo": "string",
+          "map": [
+            "object",
+            {
+              "bar": "string",
+              "id": "string"
+            }
+          ],
+          "number": "number"
+        }
+      ]
+    },
+    "list": {
+      "value": [
+        "foo",
+        "bar"
+      ],
+      "type": [
+        "tuple",
```

Comment:
Since we use list type for input variables, can we have same for output variables too?

Reply:
Yeah it's weird but it's probably just how Terraform represents list in `.tfstate` files. For example the `urls_list` in GCS config is represented as below:

```
    "urls_list": {
      "value": [
        "gs://jieyutian-test-bucket-7"
      ],
      "type": [
        "tuple",
        [
          "string"
        ]
      ]
    }
```

Does it make sense to you?

Reply:
@tjy9206 Ohkay! I am wondering if can we transform `tuple` to `list` ourselves before writing metadata. Terraform might be representing list as tuple in .tfstate because the size of list output would be fixed as the deployment has already happened. But we are adding type to metadata and before deployment there is no way of definitely knowing the size of the list output. @bharathkkb WDYT?

Reply:
It will be tricky though since we probably don't know ahead of time whether it should be a tuple of, say, two strings (key-value pair for example) or a list of strings. I would propose we keep the output generated by Terraform and have a separate post-processing. WDYT?

Reply:
I discussed this with @q2w today and we agreed to keep as is for now (without any post processing)